src/main/resources/Development-documentation/gerrit-scenarios.md

```diff
@@ -0,0 +1,316 @@
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]` (`P1`)
+- Receives comments from reviewer
+- Updates the code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]` (`P2`)
+- Gets ok from the reviewer
+- Forced to rebase the code due to `C01` (`C3`)  event `E01`
+- Pushes to `refs/for/[branch name]`  (`P3`)
+- Hits the submit button in Gerrit
+
+Eiffel events sent from the plugin:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCC(`E3`) sent for `P3` push with `PREVIOUS_VERSION` link set `E2` and `BASE` link set to `E01`
+- SCS(`E4`) sent at submit with `CHANGE` link set `E3` and `PREVIOUS_VERSION` link set to `E01`
+
+Commit history after submit:
+
+```git
+* C3 Rebased changes
+* C01 Other code change
+* C0 Commit from previous user
+```
+
+Trigger table:
+
+| Event | Triggered by | Comment |
+| ---   | ---          | ---     |
+| E0    |              | Submit  |
+| E1    | P1           |
+| E2    | P2           |
+| E3    | P3           |
+| E4    |              | Submit  |
+
+SCC Event table:
+
+| Event | BASE | PREVIOUS_VERSION |
+| ---   | ---  | ---              |
+| E1    | E0   |
+| E2    | E0   | E1               |
+| E3    | E01  | E2               |
+
+## Create patch set, submitting with rebase strategy - CASE3
+
+Scenario overview:
+
+This scenario describes a workflow where a user performs a standard review
+cycle. Before the review finishes another review finishes and submits to
+master(`C01`). Due to project setting *Rebase if necessary* and no conflicting
+changes, Gerrit will perform a rebase when the user submits the review.
+
+Preconditions:
+
+- Previous review submitted to the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]` (`P1`)
+- Receives comments from reviewer
+- Updates the code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]` (`P2`)
+- Gets ok from the reviewer
+- Hits the submit button in Gerrit
+- Gerrit will rebase (`C3`)
+
+Eiffel events sent from the plugin:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCS(`E3`) sent at submit with `CHANGE` link set `E2` and `PREVIOUS_VERSION` link set to `E01`
+
+Commit history after submit:
+
+```git
+* C3 Rebased changes
+* C01 Other code change
+* C0 Commit from previous user
+```
+
+Trigger table:
+
+| Event | Triggered by | Comment |
+| ---   | ---          | ---     |
+| E0    |              | Submit  |
+| E1    | P1           |
+| E2    | P2           |
+| E3    |              | Submit  |
+
+SCC Event table:
```

Comment:
Do we need an extra table for the SCS events also?

Reply:
As we only send one SCS event but multiple SCC I provided a table for the SCC events. Do you want a table for the SCS events?

Reply:
Maybe not, since I noticed the sentence explaining how the SCS event links, and it might not add much value with another table of only one event.